templates/shift-scheduling/schema.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"time"
+
+	"github.com/nextmv-io/sdk/mip"
+)
+
+// output holds the output data of the solution.
+type output struct {
+	AssignedShifts        []outputAssignment `json:"assigned_shifts"`
+	NumberAssignedWorkers int                `json:"number_assigned_workers"`
+}
+
+// options holds custom configuration data.
+type options struct {
+	Penalty            penalty       `json:"penalty" usage:"set penalties for over and under supply of workers"`
+	MaxHoursPerDay     time.Duration `json:"max_hours_per_day" default:"10h" usage:"maximum number of hours per day"`
+	MaxHoursPerWeek    time.Duration `json:"max_hours_per_week" default:"40h" usage:"maximum number of hours per week"`
+	MinHoursPerShift   time.Duration `json:"min_hours_per_shift" default:"2h" usage:"minimum number of hours per shift"`
+	MaxHoursPerShift   time.Duration `json:"max_hours_per_shift" default:"8h" usage:"maximum number of hours per shift"`
+	HoursBetweenShifts time.Duration `json:"hours_between_shifts" default:"8h" usage:"minimum number of hours between shifts"`
+	Limits             mip.Limits    `json:"limits" usage:"holds a field to set the maximum duration of the solver"`
```

Comment:
let's call that `solve.duration` (nested) like in nextroute

Reply:
this is because all mip model use that Limits struct. so they are all aligned (and they were with router as well, I believe). this was also a wish sebas had.